src/MuTalk-Model/FixedRandomOperatorMutantSelection.class.st

```diff
@@ -0,0 +1,32 @@
+Class {
+	#name : #FixedRandomOperatorMutantSelection,
+	#superclass : #FixedRandomMutantSelection,
+	#category : #'MuTalk-Model-Mutations generation strategies'
+}
+
+{ #category : #accessing }
+FixedRandomOperatorMutantSelection >> selectMutantsFrom: aCollection [
+
+	| size index newColl random dict operators operator |
+	newColl := aCollection copyEmpty.
+	random := Random new.
+
+	numberOfMutants = 0 ifTrue: [ ^ newColl ].
+	aCollection size <= numberOfMutants ifTrue: [ ^ aCollection ].
+
+	dict := aCollection groupedBy: [ :e | e operator ].
+	operators := dict keys.
+	1 to: numberOfMutants do: [ :i |
+		operator := operators at:
+			            (random nextIntegerBetween: 1 and: operators size).
+		[ (dict at: operator) allSatisfy: [ :each | newColl includes: each ] ]
```

Comment:
Nice check! but it would be nice it comes with a comment :)

It's important to understand that if you already selected all the mutants in an operator, then you need to filter the operator out, otherwise this will loop forever looking for mutants (and they are all used)